modules/coreos-layering-configuring-on.adoc

```diff
@@ -0,0 +1,225 @@
+
+* You have a pull secret that your nodes need to pull the new custom layered image from your registry. This should be a different secret than the one used to push the image to the repository.
+
+* You are familiar with how to configure a Containerfile. Instructions on how to create a Containerfile are beyond the scope of this documentation.
+
+* Optional: You have a separate machine config pool for the nodes where you want to apply the custom layered image.
+
+.Procedure
+
+. Create a `machineOSconfig` object:
+
+.. Create a YAML file similar to the following:
++
+[source,terminal]
+----
+apiVersion: machineconfiguration.openshift.io/v1alpha1
+kind: MachineOSConfig
+metadata:
+  name: layered
+spec:
+  machineConfigPool:
+    name: <mcp_name> <1>
+  buildInputs:
+    containerFile: # <2>
+    - containerfileArch: noarch
+      content: |-
+        FROM configs AS final
+        RUN rpm-ostree install cowsay && \
+          ostree container commit
+    imageBuilder: # <3>
+      imageBuilderType: PodImageBuilder
+    baseImagePullSecret: # <4>
+      name: global-pull-secret-copy
+    renderedImagePushspec: image-registry.openshift-image-registry.svc:5000/openshift/os-image:latest  # <5>
+    renderedImagePushSecret: # <6>
+      name: builder-dockercfg-7lzwl
+  buildOutputs: # <7>
+    currentImagePullSecret:
+      name: builder-dockercfg-7lzwl
+----
+<1> Specifies the name of the machine config pool associated with the nodes where you want to deploy the custom layered image.
+<2> Specifies the Containerfile to configure the custom layered image.
+<3> Specifies the name of the image builder to use. This must be `PodImageBuilder`.
+<4> Specifies the name of the pull secret that the MCO needs to pull the base operating system image from the registry.
+<5> Specifies the image registry to push the newly-built custom layered image to. This can be any registry that your cluster has access to. This example uses the internal {product-title} registry.
+<6> Specifies the name of the push secret that the MCO needs to push the newly-built custom layered image to that registry.
+<7> Specifies the secret required by the image registry that the nodes need to pull the newly-built custom layered image. This should be a different secret than the one used to push the image to your repository.
+
+.. Create the `MachineOSConfig` object:
++
+[source,terminal]
+----
+$ oc create -f <file_name>.yaml
+----
++
+The build process creates a `MachineOSBuilds` object and two new pods.
++
+[source,terminal]
+----
+NAME                                                              READY   STATUS    RESTARTS   AGE
+build-rendered-layered-ad5a3cad36303c363cf458ab0524e7c0           2/2     Running   0          2m40s # <1>
+# ...
+machine-os-builder-6fb66cfb99-zcpvq                               1/1     Running   0          2m42s # <2>
+----
+<1> This is the build pod where the custom layered image is building.
+<2> This pod can be used for troubleshooting.
```

Comment:
Maybe turn this into its own third (verification-like) substep, showing the `oc get pod` command used to get this output? Unless the arguments to make these 2 specific pods show up for real is too convoluted...

Making this up (with the namespace bit), but e.g.:

```suggestion

.. The build process creates a `MachineOSBuilds` object and two new pods. You can confirm that the following new pods are running:
+
[source,terminal]
----
$ oc get pods -n <machineosbuilds_namespace>
----
+
.Example output
[source,terminal]
----
NAME                                                              READY   STATUS    RESTARTS   AGE
build-rendered-layered-ad5a3cad36303c363cf458ab0524e7c0           2/2     Running   0          2m40s # <1>
# ...
machine-os-builder-6fb66cfb99-zcpvq                               1/1     Running   0          2m42s # <2>
----
<1> This is the build pod where the custom layered image is building.
<2> This pod can be used for troubleshooting.
```